openmls/src/group/mls_group/creation.rs

```diff
@@ -168,7 +168,7 @@ impl MlsGroup {
         mls_group_config: &MlsGroupConfig,
         aad: &[u8],
         credential_with_key: CredentialWithKey,
-    ) -> Result<(Self, MlsMessageOut), ExternalCommitError> {
+    ) -> Result<(Self, MlsMessageOut, Option<GroupInfo>), ExternalCommitError> {
```

Comment:
To slightly optimize things we only return a `GroupInfo` when the group to be joined has RatchetTree extension in its required capabilities

Reply:
Right, but that should always be the case for external joins, or am I missing something?

Reply:
Most of the time yes you'll do that but looking at openmls code it seems not to be mandatory. You can join by external commit [supplying a tree yourself](https://github.com/openmls/openmls/blob/0d671bd388e0d0a8167e552d8521be368090c631/openmls/src/group/mls_group/creation.rs#L172) in case the GroupInfo lacks the ratchet tree.  
That's why it could make sense to return an optional GroupInfo. Or should we deprecate this option in join_by_external_commit ?